core/src/main/java/com/rudderstack/android/sdk/core/EventRepository.java

```diff
@@ -75,6 +75,19 @@
         RudderLogger.logDebug(String.format("EventRepository: constructor: %s", this.config.toString()));
 
         try {
+            // initiate RudderPreferenceManager
+            this.preferenceManager = RudderPreferenceManager.getInstance(_application);
+            if (preferenceManager.getOptStatus()) {
+                if (_anonymousId != null) {
```

Comment:
Just set it to null. We don't need a check if it is null or not.!

Reply:
Implemented, please review it again. Thanks.